### PR TITLE
Di 27499 handle sync and pfs post race condition

### PR DESCRIPTION
**Pull request by saargt**

### CODE REVIEW

File: pfs-open-sync/src/update-cache/pns-to-cache-changes-input-converter.ts [hunk 1]
```diff
@@ -27,24 +31,35 @@ export class PnsToCacheChangesInputConverter
 		}
 	}
 
-	protected processModifiedObjects(pnsNotification: ModifiedObjectNotification): CacheObject[]
+	protected async processModifiedObjects(pnsNotification: ModifiedObjectNotification): Promise<CacheObject[]>
 	{
-		const modifiedObjects = pnsNotification.Message.ModifiedObjects;
+		// When an entry is inserted, there's no Hidden field in the ModifiedFields array.
```
Comment (Chasky-H) [inline pfs-open-sync/src/update-cache/pns-to-cache-changes-input-converter.ts:36]: Is this a real thing? 
Does the CPI sync hidden objects that haven't been synced yet?

File: pfs-open-sync/src/update-cache/pns-to-cache-changes-input-converter.ts [hunk 2]
```diff
@@ -27,24 +31,35 @@ export class PnsToCacheChangesInputConverter
...
+		// (e.g. an entry has been created AND hidden in cpi-side, then synced)
+		// We must check if the object is hidden by checking the Hidden field in the object itself.
+		// To keep the code simple, get the object's Hidden field for either insert or update.
```
Comment (Chasky-H) [inline pfs-open-sync/src/update-cache/pns-to-cache-changes-input-converter.ts:39]: getting the Hidden field for both insert or update might be simple, back also costly.
Cache PNSs should preferably not make a GET of the object and use only the PNS data when possible.

File: server-side/PfsCommands/AtomicCommands/importResourcesCommand.ts [hunk 3]
```diff
@@ -70,6 +70,17 @@ export class ImportResourcesCommand extends PfsService implements ICommand
 	{
 		const existingFile: AddonFile = existingFilesMap.get(obj.Object.Key) ?? { doesFileExist: false };
 
+		// Check if the object has been uploaded
```
Comment (Chasky-H) [inline server-side/PfsCommands/AtomicCommands/importResourcesCommand.ts:73]: Please look into alternative solutions that don't generate errors and advise.

File: pfs-open-sync/src/update-cache/pns-to-cache-changes-input-converter.ts (hunk 1 above)
Comment (saargt) [reply pfs-open-sync/src/update-cache/pns-to-cache-changes-input-converter.ts:36]: It is a real thing in my tests (I insert offline, then delete offline, then sync).
I can imagine such a flow in reality as well - Create a Survey with an image, then delete said survey and image. Later on sync is performed.

**Comment by Chasky-H** [comment]
Please open tasks for outstanding comments. Otherwise you can merge.

**Review by Chasky-H — APPROVED** [review]

**Pull request merged**